packages/atomic/src/components/commerce/atomic-commerce-products-per-page/atomic-commerce-products-per-page.spec.ts

```diff
@@ -0,0 +1,348 @@
+
+describe('AtomicCommerceProductsPerPage', () => {
+  const mockedEngine = buildFakeCommerceEngine();
+  const mockedFocusOnFirstResultAfterNextSearch = vi.fn(() =>
+    Promise.resolve()
+  );
+  const mockedFocusOnNextNewResult = vi.fn();
+  let mockedSummary: Summary;
+  let mockedPager: Pagination;
+
+  const renderProductsPerPage = async ({
+    interfaceElementType = 'search',
+    paginationState = {},
+    summaryState = {},
+    props = {},
+  }: {
+    interfaceElementType?: 'product-listing' | 'search';
+    paginationState?: Partial<PaginationState>;
+    summaryState?: Partial<SearchSummaryState | ProductListingSummaryState>;
+    props?: {
+      choicesDisplayed?: string;
+      initialChoice?: number;
+    };
+  } = {}) => {
+    mockedSummary = buildFakeSummary({state: summaryState});
+
+    mockedPager = buildFakePager({
+      state: paginationState,
+    });
+
+    vi.mocked(buildProductListing).mockReturnValue(
+      buildFakeProductListing({
+        implementation: {
+          pagination: () => mockedPager,
+          summary: () => mockedSummary,
+        },
+      })
+    );
+    vi.mocked(buildSearch).mockReturnValue(
+      buildFakeSearch({
+        implementation: {
+          pagination: () => mockedPager,
+          summary: () => mockedSummary as Summary<SearchSummaryState>,
+        },
+      })
+    );
+
+    const {element} =
+      await renderInAtomicCommerceInterface<AtomicCommerceProductsPerPage>({
+        template: html`<atomic-commerce-products-per-page
+          choices-displayed=${ifDefined(props.choicesDisplayed)}
+          initial-choice=${ifDefined(props.initialChoice)}
+        ></atomic-commerce-products-per-page>`,
+        selector: 'atomic-commerce-products-per-page',
+        bindings: (bindings) => {
+          bindings.interfaceElement.type = interfaceElementType;
+          bindings.engine = mockedEngine;
+          bindings.store.state.resultList = {
+            focusOnFirstResultAfterNextSearch:
+              mockedFocusOnFirstResultAfterNextSearch,
+            focusOnNextNewResult: mockedFocusOnNextNewResult,
+          };
+          return bindings;
+        },
+      });
+
+    return {
+      element,
+      get legend() {
+        return element.shadowRoot!.querySelector('legend');
+      },
+      get label() {
+        return element.shadowRoot!.querySelector('span[part="label"]');
+      },
+      get buttonList() {
+        return element.shadowRoot!.querySelector('div[part="buttons"]');
+      },
+      get buttons() {
+        return element.shadowRoot!.querySelectorAll('input[part*="button"]');
+      },
+      get activeButton() {
+        return element.shadowRoot!.querySelector(
+          'input[part*="active-button"]'
+        );
+      },
+    };
+  };
+
+  beforeEach(() => {
+    vi.spyOn(console, 'error').mockImplementation(() => {});
+  });
+
+  it('should log an error and throw if choicesDisplayed is not a valid comma-separated list of numbers', async () => {
```

Comment:
IMO the proper way to test the errors would be to verify that we're calling `convertChoicesToNumbers` and `validateInitialChoice` with the correct args in the initialization, and test those exported util functions separately (in a validate.spec.ts file).